common/src/main/java/com/revenuecat/purchases/common/verification/SigningManager.kt

```diff
@@ -63,12 +63,12 @@ class SigningManager(
         return if (verificationResult) {
             VerificationResult.SUCCESS
         } else {
-            errorLog(NetworkStrings.VERIFICATION_ERROR)
+            errorLog(NetworkStrings.VERIFICATION_ERROR.format(urlPath))
             VerificationResult.ERROR
         }
     }
 
     private fun getSignatureMessage(responseCode: Int, body: String?, eTag: String?): String? {
-        return if (responseCode == RCHTTPStatusCodes.NOT_MODIFIED) eTag else body
+        return if (responseCode == RCHTTPStatusCodes.NOT_MODIFIED) eTag else body?.let { "$it\n" }
```

Comment:
I thought about adding a test making sure this happens, but it's tricky since this all happens internally to this class and it seemed wrong to move that logic away from this class. I thought with the real data test would be enough to cover this case.

Reply:
Hmm why adding a line break? Responses always end with it, and if they stop adding it, then the signature needs to account for that.
In iOS I added the line break to the test to replicate what the real responses look like so the signature matches.

Reply:
Well, the way we are [reading the body](https://github.com/RevenueCat/purchases-android/blob/main/common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt#L60-L67) in android (per lines) actually removes that line break. Which was causing the signature verification to fail. Adding this line break at the end actually fixes the issue (it verifies correctly). 

Having said that, you might be right... It might be more accurate to change how we read the body to avoid modifying it at all... Will work on that tomorrow. Thanks!

Reply:
👍🏻

Reply:
Changed approach in latest commits @NachoSoto. Lmk what you think!